tests/unit/Alignment.cpp

```diff
@@ -0,0 +1,355 @@
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <bit>
+
+#define BOOST_TEST_MODULE TypeAlignment
+bool init_unit_test();
+#include "boost/test/unit_test.hpp"
+
+#include "revng/ABI/Definition.h"
+#include "revng/ADT/Concepts.h"
+#include "revng/Model/Binary.h"
+
+static std::string printAlignment(std::uint64_t Alignment) {
+  return Alignment != 0 ? std::to_string(Alignment) : "undefined";
+}
+
+struct Expected {
+  const abi::Definition &ABI;
+  std::uint64_t Alignment;
+
+  explicit Expected(model::ABI::Values ABIName, std::uint64_t Alignment) :
+    ABI(abi::Definition::get(ABIName)), Alignment(Alignment) {}
+};
+
+struct ConstantTest {
+  model::QualifiedType Type;
+
+  // clang-format off
+  template<typename ...Types>
+    requires(same_as<Types, Expected> && ...)
+  void run(const Types &...TestCases) {
+    // clang-format on
+    for (auto [ABI, Expected] : std::array{ TestCases... }) {
+      std::optional<std::uint64_t> TestResult = ABI.alignment(Type);
+      if (TestResult.value_or(0) != Expected) {
+        std::string Error = "Alignment run failed for type:\n"
+                            + serializeToString(Type) + "ABI ('"
+                            + serializeToString(ABI.ABI())
+                            + "') reports the alignment of '"
+                            + printAlignment(TestResult.value_or(0)) + "', "
+                            + "while the expected value is '"
+                            + printAlignment(Expected) + "'.\n";
+        revng_abort(Error.c_str());
+      }
+    }
+  }
+};
```

Comment:
This can become a simple function now. The `run` method is always invoked once AFAIU.

Reply:
Done.